### DISCUSSION TITLE
Allocated memory

### DISCUSSION THREAD

**Discussion by marcowhites** [modflowpy/flopy#1386] [General]
Hi,
I am running multiple instances of a big transient MF6 model within Flopy on a HPC and I am encountering some issue with amount of memory required by Flopy and the number of nodes I can use. I wonder if others have had the same issue. Is there a way to free up RAM after the write_simulation() command. I can run the MF6 outside Python, but I wonder if this is the only way. Does the MFSimulation object (flopy.mf6.modflow.mfsimulation) contain all the data and therefore becomes bigger and bigger as we add data for the MF6 packages? 

Thanks 
Marco Bianchi (BGS)

**Comment by christianlangevin** [comment]
@spaulins-usgs and @aleaf  will probably have some ideas, but one thing you might want to try is to use external files, and binary if possible.  Flopy won't try to load them unless it needs to.

**Reply by marcowhites** [reply]
Thanks Chris (we met in Alabama long time ago when I was doing my PhD). 
From what you said, I suspect that the main problem may be the stress period data of the recharge package which is very large (large model and lot of stress periods)

**Comment by spaulins-usgs** [comment]
@marcowhites, the MFSimulation object directly or indirectly contains all the data (MFSimulation contains MFModel and MFPackage objects which contain MFPackage, MFBlock, and MFData objects).  @langevin-usgs's suggestion of storing the data in external files should help.  If you are not already doing so, I also recommend using a 64 bit version of python instead of a 32 bit version, which should give you a higher maximum of allocated memory.

@langevin-usgs, I could possibly add an "unload_data" method to MFSimulation, which unloads all data from memory except what is necessary to run the simulation.  The MFSimulation object would then be in an "unloaded" state, and would either return appropriate error messages if the user tried to access data or perhaps automatically reload the simulation when data access is attempted.

**Reply by marcowhites** [reply]
Thanks for the answer. 
I am trying to modify my Flopy script to use external data (basically I want to implement the "open/close" option for the block PERIOD in the *.rch input file. What would the format  of the stress period data be in this case? Just a list of file names? 
I found this example https://github.com/modflowpy/flopy/issues/879 but I am not sure.

Thanks again

**Reply by spaulins-usgs** [reply]
@macrowhites, for examples please look at our flopy for modflow-6 tutorial notebook here:

https://github.com/modflowpy/flopy/blob/develop/examples/Notebooks/flopy3_mf6_tutorial.ipynb

There is an example in the subsection "Specifying MFList Data in an External File".  You will probably want to modify the example so that it just points to the data file instead of creating the data file.  Your stress period data for the first two stress periods would therefore look something like this:

    stress_period_data = {
        0: {"filename": "chd_sp1.dat"},
        1: {"filename": "chd_sp2.dat"},
    }
